File: backend/src/main/java/com/form/org/Services/StockService.java
package com.form.org.Services;

import com.form.org.dto.StockDTO;
import com.form.org.model.Stock;

import java.math.BigDecimal;
import java.util.Date;
import java.util.List;
import java.util.Optional;

public interface StockService {
    StockDTO save(StockDTO dto);

    StockDTO findById(Integer idStock);
    BigDecimal calculateStock(Integer idArticle);
    StockDTO findBydateStock(Date dateStock);

    List<StockDTO> findAll();

    void delete(Integer idStock);
}

File: FrontEnd1/src/app/services/stock/stock.service.ts
import { Injectable } from '@angular/core';
import { Observable, of } from 'rxjs';
import { StockDTO } from 'src/gs-api/src/models/stock-dto';
import { StockService } from 'src/gs-api/src/services';

@Injectable({
  providedIn: 'root'
})
export class StockServices {


  constructor(
    private stockservice: StockService)
     { }
     
   enregistrerStock(stockDt:StockDTO):Observable<StockDTO> {
    return this.stockservice.save(stockDt);
  }
  findAllStock(): Observable<StockDTO[]> {
   return this.stockservice.findAll();
 }

 findStockById(idStock?: number): Observable<StockDTO> {
   if (idStock) {
     return this.stockservice.findById(idStock);
   }
   return of();
 }

 deleteStock(idStock?: number):Observable<any>{
   if (idStock) {
     return this.stockservice.delete(idStock);
   }
   return of();
 }

}


